src/device/mod.rs

```diff
@@ -1465,12 +1468,37 @@ impl Runtime {
             .map(|peers| peers.iter().any(|p| p.public_key == exit_node.public_key))
             .unwrap_or_default();
 
+        self.requested_state.postquantum_wg = None;
+
         if is_meshnet_exit_node {
             if let Some(dns) = &self.entities.dns.lock().await.resolver {
                 self.reconfigure_dns_peer(dns, &dns.get_default_dns_servers())
                     .await?;
             }
-        } else if exit_node.endpoint.is_none() {
+        } else if let Some(addr) = exit_node.endpoint {
```

Comment:
Also note, that we should handle meshnet<->PQ-WG incompatabilities, by either erroring out in an incompatible situation, or reconnecting. It would be nice to sync sync with Linux-app devs on exact solution.

Reply:
Yeah, since PQ is a feature, I decided to throw an error from `set_config()` when PQ is enabled https://github.com/NordSecurity/libtelio/pull/259/commits/11dcb11051004f339f7947e19b965aa64f2e74c5
I've planned to merge it with the follow-up PR, I can port this commit to this one

Reply:
If it is planned in follow up PR -> it is ok, we can add it there.

Reply:
I've added it here